kolibri/plugins/user_profile/assets/src/views/ChangeFacility/index.vue

```diff
@@ -69,22 +70,31 @@
     created() {
       this.service.start();
       this.service.onTransition(state => {
-        const stateID = Object.keys(state.meta)[0];
-        if (state.meta[stateID] !== undefined) {
-          let newRoute = state.meta[stateID].route;
-          if (newRoute != this.$router.currentRoute.name) {
-            if ('path' in state.meta[stateID]) {
-              this.internalNavigation = true;
-              this.$router.push(
-                { name: newRoute, path: state.meta[stateID].path },
-                function() {
-                  this.internalNavigation = false;
-                }.bind(this)
-              );
-            } else this.$router.push(newRoute);
+        if (state.value === 'error') {
+          this.$store.commit(
+            'CORE_SET_ERROR',
+            `An error occured in the '${this.previousMachineStateName}' state of the change facility machine`
+          );
+          this.service.send('RESET');
+        } else {
+          const stateID = Object.keys(state.meta)[0];
+          if (state.meta[stateID] !== undefined) {
```

Comment:
@MisRob I can't seem to understand whats going on in this block, furthermore, the deep nesting makes it harder to read and follow. I think refactoring this into function(s) could make this more clearer

Reply:
I think this connects the machine with the router. You're right that I added one level of nesting with my change, so I'll update this whole part to improve readability and also comment on the part you pointed out on that opportunity in more detail.

Reply:
Thanks @MisRob. My apologies, the comment is indend for the while block line:[79](https://github.com/learningequality/kolibri/pull/9715/files#diff-57b8e61916c5f184c9b4a498234a098481a08861a0b852b5e898153fe96699c4R79) to [96](https://github.com/learningequality/kolibri/pull/9715/files#diff-57b8e61916c5f184c9b4a498234a098481a08861a0b852b5e898153fe96699c4R97)

Reply:
I think we will need to hear from @jredrejo because I don't recall what was the reason for that block exactly

Reply:
Regarding additions in this PR, pushed a commit to simplify it